OutboundCallReminder/src/main/java/com/communication/outboundcallreminder/App.java

```diff
@@ -0,0 +1,171 @@
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import com.azure.communication.common.CommunicationUserIdentifier;
+import com.azure.communication.identity.CommunicationIdentityClient;
+import com.azure.communication.identity.CommunicationIdentityClientBuilder;
+import com.communication.outboundcallreminder.Ngrok.NgrokService;
+import com.microsoft.cognitiveservices.speech.SpeechConfig;
+import com.microsoft.cognitiveservices.speech.SpeechSynthesisOutputFormat;
+import com.microsoft.cognitiveservices.speech.SpeechSynthesizer;
+import com.microsoft.cognitiveservices.speech.audio.AudioConfig;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class App {
+    private static NgrokService ngrokService;
+    final static String url = "http://localhost:9007";
+    final static String serverPort = "9007";
+
+    public static void main(String[] args) {
+        SpringApplication application = new SpringApplication(App.class);
+        application.setDefaultProperties(Collections.singletonMap("server.port", serverPort));
+        application.run(args);
+
+        Logger.LogMessage(Logger.MessageType.INFORMATION, "Starting ACS Sample App ");
+
+        // Get configuration properties
+        ConfigurationManager configurationManager = ConfigurationManager.GetInstance();
+        configurationManager.LoadAppSettings();
+
+        // Start Ngrok service
+        String ngrokUrl = StartNgrokService();
+        try {
+            if (ngrokUrl != null && !ngrokUrl.isEmpty()) {
+                Logger.LogMessage(Logger.MessageType.INFORMATION,"Server started at -- > " + url);
+                Thread runSample = new Thread(() -> {
+                    RunSample(ngrokUrl);
+                });
+                runSample.start();
+                runSample.join();
+            } else {
+                Logger.LogMessage(Logger.MessageType.INFORMATION,"Failed to start Ngrok service");
+            }
+        } catch (Exception ex) {
+            Logger.LogMessage(Logger.MessageType.ERROR,"Failed to start Ngrok service -- > " + ex.getMessage());
+        }
+        Logger.LogMessage(Logger.MessageType.INFORMATION, "Press 'Ctrl + C' to exit the sample");
+        ngrokService.Dispose();
+    }
+
+    private static String StartNgrokService() {
+        try {
+            ConfigurationManager configurationManager = ConfigurationManager.GetInstance();
+            String ngrokPath = configurationManager.GetAppSettings("NgrokExePath");
+
+            if (ngrokPath.isEmpty()) {
+                Logger.LogMessage(Logger.MessageType.INFORMATION, "Ngrok path not provided");
+                return null;
+            }
+
+            Logger.LogMessage(Logger.MessageType.INFORMATION,"Starting Ngrok");
+            ngrokService = new NgrokService(ngrokPath, null);
```

Comment:
Looks like we are passing null for ngrok authToken. Don't we need to pass a valid value, possibly read from config?

Reply:
Not required, at least in our samples case. As per ngrok docs, authtoken require for some advanced features.